common/djangoapps/django_comment_common/models.py

```diff
@@ -105,3 +106,29 @@ class Meta:
 
     def __unicode__(self):
         return self.name
+
+
+def all_permissions_for_user_in_course(user, course_id):  # pylint: disable=invalid-name
+    """Returns all the permissions the user has in the given course."""
+    course = modulestore().get_course(course_id)
+    if course is None:
+        raise ItemNotFoundError(course_id)
+    course_blacked_out = not course.forum_posts_allowed
+
+    all_roles = {role.name for role in Role.objects.filter(users=user, course_id=course_id)}
+
+    def permission_blacked_out(p_name):
+        """Returns true if the given permission is blacked out"""
+        return (
+            course_blacked_out and
+            all_roles == {FORUM_ROLE_STUDENT} and
+            (p_name.startswith('edit') or p_name.startswith('update') or p_name.startswith('create'))
```

Comment:
Nit: this seems pretty brittle. I wonder if a property could be added to the permissions to indicate that they are not allowed on "blacked out" courses.

This doesn't need to hold up the PR though-- we know the discussion permissions need further work.